web/src/main/java/org/cbioportal/weblegacy/MutationalSignatureController.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2016 Memorial Sloan-Kettering Cancer Center.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY, WITHOUT EVEN THE IMPLIED WARRANTY OF MERCHANTABILITY OR FITNESS
+ * FOR A PARTICULAR PURPOSE. The software and documentation provided hereunder
+ * is on an "as is" basis, and Memorial Sloan-Kettering Cancer Center has no
+ * obligations to provide maintenance, support, updates, enhancements or
+ * modifications. In no event shall Memorial Sloan-Kettering Cancer Center be
+ * liable to any party for direct, indirect, special, incidental or
+ * consequential damages, including lost profits, arising out of the use of this
+ * software and its documentation, even if Memorial Sloan-Kettering Cancer
+ * Center has been advised of the possibility of such damage.
+ */
+
+/*
+ * This file is part of cBioPortal.
+ *
+ * cBioPortal is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+package org.cbioportal.weblegacy;
+
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import java.util.List;
+import org.cbioportal.model.MutationalSignature;
+import org.cbioportal.service.MutationalSignatureService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.bind.annotation.*;
+
+/**
+ *
+ * @author heinsz
+ */
+
+@RestController
+public class MutationalSignatureController {
+
+    @Autowired
+    private MutationalSignatureService mutationalSignatureService;
+
+    @ApiOperation(value = "Get mutation signatures for given samples in a study gene panel information",
+            nickname = "getGenePanel",
+            notes = "")
+    @Transactional
+    @RequestMapping(method = {RequestMethod.GET, RequestMethod.POST}, value = "/mutational-signature")
```

Comment:
we messaged a bit about GET v. POST and when to use one or the other. This is in weblegacy, and the "handle both request types" was normal in ApiController. We should have a bigger discussion as a development team about how to use request types in the REST API